File: e-livraria-web/src/models/PromotionalCoupon.ts

export interface IPromotionalCoupon{
    id?: number;
    name?: string;
    value?: number;
}

export class PromotionalCoupon{
    public id: number;
    public name: string;
    public value: number;

    constructor( {
        id = 0,
        name = '',
        value = 0,
    }: IPromotionalCoupon){
        this.id = id;
        this.name = name;
        this.value = value;
    }

    // get id(){
    //     return this._id;
    // }
    // set id(id: number){
    //     this._id = id;
    // }

    // get name(){
    //     return this._name;
    // }
    // set name(name: string){
    //     this._name = name;
    // }

    // get value(){
    //     return this._value;
    // }
    // set value(value: number){
    //     this._value = value;
    // }
}

File: ecommerceLivraria/src/main/java/com/nrabello/ecommerceLivraria/model/PromotionalCoupon.java
package com.nrabello.ecommerceLivraria.model;

import com.fasterxml.jackson.annotation.JsonIgnore;
import jakarta.persistence.*;

import java.util.HashSet;
import java.util.Set;

@Entity
@Table(name = "Cupons_Promocionais")
public class PromotionalCoupon {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name="cup_id")
    private Long id;

    @Column(name="cup_nome")
    private String name;

    @Column(name="cup_valor")
    private Double value;

//    @OneToMany()
//    @JsonIgnore
//    private Set<Order> orders = new HashSet<>();

    public PromotionalCoupon(){

    }
    public PromotionalCoupon(Long id, String name, Double value, Set<Order> orders) {
        this.setId(id);
        this.setName(name);
        this.setValue(value);
//        this.setOrders(orders);
    }

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public Double getValue() {
        return value;
    }

    public void setValue(Double value) {
        this.value = value;
    }

//    public Set<Order> getOrders() {
//        return orders;
//    }
//
//    public void setOrders(Set<Order> orders) {
//        this.orders = orders;
//    }
}
